pkg/scheduler/core/spreadconstraint/select_clusters_by_region.go

```diff
@@ -35,66 +36,131 @@ func selectBestClustersByRegion(spreadConstraintMap map[policyv1alpha1.SpreadFie
 	}
 
 	// thirdly, select the remaining Clusters based cluster.Score
-	sortClusters(candidateClusters)
 	restCnt := needCnt - len(clusters)
-	for i := 0; i < restCnt; i++ {
-		clusters = append(clusters, candidateClusters[i].Cluster)
+	if restCnt > 0 {
+		sortClusters(candidateClusters)
+		for i := 0; i < restCnt; i++ {
+			clusters = append(clusters, candidateClusters[i].Cluster)
+		}
 	}
 
 	return clusters, nil
 }
 
-func selectRegions(RegionInfos map[string]RegionInfo, regionConstraint, clusterConstraint policyv1alpha1.SpreadConstraint) []RegionInfo {
-	var regions []RegionInfo
-	for i := range RegionInfos {
-		regions = append(regions, RegionInfos[i])
+// getSortedRegions returns sorted region slice by score and available replicas.
+func getSortedRegions(regionInfoMap map[string]RegionInfo) []RegionInfo {
+	regions := make([]RegionInfo, 0, len(regionInfoMap))
+	for _, region := range regionInfoMap {
+		regions = append(regions, region)
 	}
 
 	sort.Slice(regions, func(i, j int) bool {
 		if regions[i].Score != regions[j].Score {
 			return regions[i].Score > regions[j].Score
 		}
 
+		if regions[i].AvailableReplicas != regions[j].AvailableReplicas {
+			return regions[i].AvailableReplicas > regions[j].AvailableReplicas
```

Comment:
Can you develop a `score` scheduler plugin based on cluster `AvailableReplicas`? This way we don't need to deal with AvailableReplicas here.

Reply:
It's a good idea! I think this is a new feature and I'd love to work on it, but I don't think it should be included in this PR.
@RainbowMango How do you think of this?

Reply:
I'll get back to this tomorrow.

Reply:
I think the code is a bit redundant in https://github.com/karmada-io/karmada/pull/3287, and the score result is best kept in the [0,100] interval. Maybe we need to discuss it further.